utils/src/logging.rs

```diff
@@ -4,14 +4,12 @@ use tracing_subscriber::{self, fmt::format::FmtSpan, EnvFilter};
 
 pub fn setup_logging() -> Result<()> {
     let subscriber = tracing_subscriber::fmt()
-        .json()
```

Comment:
IMO we should stick to JSON output, or at least make this configurable.

To summarize my opinion:

- The nodes will not be operated by the end users, so easy of readability is not important. This stack will be used either by node operators in production, or people building on top of Miden for testing.
- For production:
    - We never want to read the logs by hand, we want to collect them via some agent and index it in a full text search db (something like logstash + elastic)
    - To make searching over the data easier, we want to parse the data, and have metadata handy for that, JSON is the simplest format for that.
- For development:
    - People will not be developing the node itself, but on top of it, so likely devs will only go over the logs when a failure happens and the node stops working, in this case they will be looking for ERROR and not inspecting dozens of pages of logs, the format is not super important here (specially given the target audience are devs, and they should be comfortable with JSON anyways)
- Lastly there are us:
    - We are most likely the only group of people that will extensively inspect logs, and to do that searching and filtering is a must. Having JSON is IMO a must, because we can use tools like `jq`, otherwise we are left to 1. running elastic search locally 2. grepping megabytes of logs. Neither of which are great. JSON is a simple solution for that

Reply:
I think it's fine to use JSON once we have a proper methodology/tools in place to parse through the logs. I was debugging the node last week, and following simple logs was pretty easy - but not sure how far I would have gotten if it was all JSON output.

Also, making it configurable is probably not too difficult - might be a good option to have in general.